src/components/ui/table/table.tsx

```diff
@@ -0,0 +1,56 @@
+import { type ReactNode } from "react"
+
+interface TableProps<T extends { id: number }> {
+  data: T[]
+  columns: string[]
+  renderCell: (item: T, column: string) => ReactNode
+  onClickRow?: (id: number) => void
+}
+
+export function Table<T extends { id: number }>({
+  data,
+  columns,
+  renderCell,
+  onClickRow,
+}: TableProps<T>): JSX.Element {
+  const handleRowClick = (item: T) => {
+    if (onClickRow != null) {
+      onClickRow(item.id)
+    }
+  }
+  return (
+    <div className='relative overflow-x-auto shadow-md sm:rounded-lg'>
+      <table className='w-full text-left rtl:text-right'>
+        <thead className='text-black'>
+          <tr>
+            {columns.map((column, index) => (
+              <th key={index} scope='col' className='pr-3 py-2 whitespace-nowrap'>
+                {column}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((item, index) => (
+            <tr
+              key={index}
+              className='bg-white border-b hover:bg-slate-100 text-black hover:cursor-pointer'
```

Comment:
Not all rows in the existing tables are clickable, maybe we should set a flag to check if it's clickable or not.

Reply:
Also, remove `hover:` for `cursor-pointer`

Reply:
> Also, remove `hover:` for `cursor-pointer`

Up on this

Reply:
Oppss, i have removed it now